Client/mods/deathmatch/logic/CClientPrimitiveBufferManager.cpp

```diff
@@ -0,0 +1,81 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *               (Shared logic for modifications)
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        mods/shared_logic/CClientPrimitiveBufferManager.cpp
+ *  PURPOSE:     Primitive buffer manager class
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+
+using std::list;
+
+CClientPrimitiveBufferManager::CClientPrimitiveBufferManager(CClientManager* pManager)
+{
+    m_pManager = pManager;
+}
+
+CClientPrimitiveBufferManager::~CClientPrimitiveBufferManager()
+{
+    DeleteAll();
+}
+
+CClientPrimitiveBuffer* CClientPrimitiveBufferManager::Get(unsigned long ulID)
+{
+    // Find the PrimitiveBuffer with the given id
+    list<CClientPrimitiveBuffer*>::const_iterator iter = m_List.begin();
+    for (; iter != m_List.end(); iter++)
+    {
+        if ((*iter)->GetID() == ulID)
+        {
+            return *iter;
+        }
+    }
+
+    return NULL;
+}
+
+CClientPrimitiveBuffer* CClientPrimitiveBufferManager::Create()
+{
+    CClientPrimitiveBuffer* pPrimitiveBuffer = new CClientPrimitiveBuffer(m_pManager, INVALID_ELEMENT_ID);
+    AddToList(pPrimitiveBuffer);
+    return pPrimitiveBuffer;
+}
+
+void CClientPrimitiveBufferManager::Delete(CClientPrimitiveBuffer* pPrimitiveBuffer)
+{
+    RemoveFromList(pPrimitiveBuffer);
+    delete pPrimitiveBuffer;
+}
+
+void CClientPrimitiveBufferManager::DeleteAll()
+{
+    list<CClientPrimitiveBuffer*>::const_iterator iter = m_List.begin();
```

Comment:
Bonus: If `m_List` contains unique_ptrs only, this stuff isn't required, as `m_List.clear()` will delete everything automatically